cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureCloudResourceService.java

```diff
@@ -187,10 +187,11 @@ public List<CloudResource> getInstanceCloudResources(String stackName, List<Clou
                                 .stream()
                                 .filter(cloudInstance -> cloudInstance.getTemplate().getStatus().equals(InstanceStatus.CREATE_REQUESTED))
                                 .filter(cloudInstance -> instance.getName().equalsIgnoreCase(
-                                        azureUtils.getPrivateInstanceId(stackName,
+                                        azureUtils.getFullInstanceId(stackName,
                                                 cloudInstance.getTemplate().getGroupName(),
-                                                Long.toString(cloudInstance.getTemplate().getPrivateId()))))
-                                .peek(cloudInstance -> LOGGER.debug("The following resource is categorized as VM instance {}", cloudInstance.toString()))
+                                                Long.toString(cloudInstance.getTemplate().getPrivateId()),
+                                                cloudInstance.getDbIdOrDefaultIfNotExists())))
+                                .peek(cloudInstance -> LOGGER.debug("The following resource is categorized as VM instance {}", cloudInstance))
```

Comment:
wouldn't this generate a lot of loglines? maybe it would worth to create a collection here, log that one, and run the foreach on that collection

Reply:
Hmm, I did not notice this, but you are right. It's not a big deal to improve this, added it to the commit, thanks!